src/modules/sync/hooks/use-sync-dialog-texts.tsx

```diff
@@ -1,46 +1,103 @@
-import { sprintf } from '@wordpress/i18n';
 import { useI18n } from '@wordpress/react-i18n';
 import { useMemo } from 'react';
-import { getEnvironmentLabel } from 'src/modules/sync/lib/environment-utils';
+import { EnvironmentType } from 'src/modules/sync/lib/environment-utils';
 
-export const useSyncDialogTexts = ( type: 'pull' | 'push', envType: string ) => {
+type TextByEnvironment = {
+	[ key in EnvironmentType ]: {
+		title: string;
+		description: string;
+	};
+};
+
+type SyncDialogTexts = {
+	title: string;
+	description: string;
+	fromLabel: string;
+	toLabel: string;
+	subtitleSelector: string;
+	envSync: string;
+	submit: string;
+};
+
+export const useSyncDialogTexts = (
+	type: 'pull' | 'push',
+	siteEnv: EnvironmentType
+): SyncDialogTexts => {
 	const { __ } = useI18n();
 
 	return useMemo( () => {
-		const envLabel = getEnvironmentLabel( envType );
-
 		if ( type === 'pull' ) {
+			const textByEnvironment: TextByEnvironment = {
+				production: {
+					title: __( 'Pull from Production' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your production site. Unchecked items will not be changed."
+					),
+				},
+				staging: {
+					title: __( 'Pull from Staging' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your staging site. Unchecked items will not be changed."
+					),
+				},
+				development: {
+					title: __( 'Pull from Development' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your development site. Unchecked items will not be changed."
+					),
+				},
+				local: {
+					title: __( 'Pull from remote Local' ),
+					description: __(
+						"Pulling will overwrite your Studio site's selected files and database with a copy from your local site. Unchecked items will not be changed."
```

Comment:
What about using the word `remote` in the description as well?

```suggestion
						"Pulling will overwrite your Studio site's selected files and database with a copy from your remote local site. Unchecked items will not be changed."
```

Reply:
Great suggestion. Done https://github.com/Automattic/studio/pull/1648/commits/be8bde22e837d7a547ed85575a3a66c2d228c0a1